app/controllers/CheckYourAnswersController.scala

```diff
@@ -65,23 +66,30 @@ class CheckYourAnswersController @Inject()(
 
       val summaryLists = getAllSummaryLists(request, businessSummaryList, salesFromEuSummaryList, waypoints)
 
-        Future(Ok(view(
-          waypoints,
-          summaryLists,
-          request.userAnswers.period,
-          totalVatToCountries,
-          totalVatOnSales,
-          noPaymentDueCountries,
-          containsCorrections,
-          errors.map(_.errorMessage))))
+      Future(Ok(view(
+        waypoints,
+        summaryLists,
+        request.userAnswers.period,
+        totalVatToCountries,
+        totalVatOnSales,
+        noPaymentDueCountries,
+        containsCorrections,
+        errors.map(_.errorMessage))))
   }
 
   def onSubmit(waypoints: Waypoints, incompletePromptShown: Boolean): Action[AnyContent] = cc.authAndRequireData().async {
     implicit request =>
-
-      coreVatReturnService.submitCoreVatReturn(request.userAnswers).map {
-        case true => Redirect(controllers.submissionResults.routes.SuccessfullySubmittedController.onPageLoad().url)
-        case _ => Redirect(controllers.submissionResults.routes.ReturnSubmissionFailureController.onPageLoad().url)
+      val userAnswers = request.userAnswers
+      coreVatReturnService.submitCoreVatReturn(userAnswers).flatMap { remainingTotalAmountVatDueGBP =>
```

Comment:
Original submitCoreVatReturn logic could never return false. Now returns the remaining amount